specification/web/Web.Management/StaticSiteARMResource.tsp

```diff
@@ -0,0 +1,400 @@
+      @path
+      userid: string;
+    },
+    OverrideErrorType = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for Updates a user entry with the listed roles
+   */
+  @summary("Updates a user entry with the listed roles")
+  @patch(#{ implicitOptionality: true })
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.Web/staticSites/{name}/authproviders/{authprovider}/users/{userid}")
+  updateStaticSiteUser is StaticSiteUserOps.ActionSync<
+    StaticSiteARMResource,
+    StaticSiteUserARMResource,
+    ArmResponse<StaticSiteUserARMResource>,
+    Parameters = {
+      #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+      @path
+      authprovider: string;
+
+      #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+      @path
+      userid: string;
+    },
+    OverrideErrorType = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for Creates or updates the app settings of a static site.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-put-operation-response-codes" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @summary("Creates or updates the app settings of a static site.")
+  @put
+  @action("config/appsettings")
+  createOrUpdateStaticSiteAppSettings is ArmResourceActionSync<
+    StaticSiteARMResource,
+    StringDictionary,
+    ArmResponse<StringDictionary>,
+    Error = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for Creates or updates the function app settings of a static site.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-put-operation-response-codes" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @summary("Creates or updates the function app settings of a static site.")
+  @put
+  @action("config/functionappsettings")
+  createOrUpdateStaticSiteFunctionAppSettings is ArmResourceActionSync<
+    StaticSiteARMResource,
+    StringDictionary,
+    ArmResponse<StringDictionary>,
+    Error = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for Creates an invitation link for a user with the role
+   */
+  @summary("Creates an invitation link for a user with the role")
+  @action("createUserInvitation")
+  createUserRolesInvitationLink is ArmResourceActionSync<
+    StaticSiteARMResource,
+    StaticSiteUserInvitationRequestResource,
+    ArmResponse<StaticSiteUserInvitationResponseResource>,
+    Error = DefaultErrorResponse
+  >;
+
+  /**
+   * Description for Detaches a static site.
+   */
+  @summary("Detaches a static site.")
+  @action("detach")
+  detachStaticSite is ArmResourceActionAsync<
+    StaticSiteARMResource,
+    void,
+    {
+      @body body: void;
+    },
```

Comment:
Why `@body: void`? What's the difference from no body?

Reply:
If we don't write this way, we could pass in `OkResponse` to represent 200 response. However, it will have a model `OkResponse` generated in swagger, and the finalresult is incorrect in the TCGC output. So our decision is `@body: void`

Reply:
It needs
```
 LroHeaders = Azure.Core.Foundations.RetryAfterHeader &
      ArmLroLocationHeader<FinalResult = void>
```